### PR TITLE
[log-shipper] Label filter

### PR DESCRIPTION
**Pull request by nabokihms**
Signed-off-by: m.nabokikh <maksim.nabokikh@flant.com>

## Description
* Add label filter support
* Add docs and examples
* Code refactoring

## Why do we need it, and what problem does it solve?
Closes https://github.com/deckhouse/deckhouse/issues/1343
Closes https://github.com/deckhouse/deckhouse/issues/1395

## Changelog entries

```changes
section: log-shipper
type: feature
summary: Add label filter support for log-shipper. Users will be able to filter log messages based on their metadata labels.
impact_level: default
```

### CODE REVIEW

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @nabokihms. I've started the workflow for label 'deploy/web/test'...

File: shell_lib/hooks/storage_class_change.sh [hunk 1]
```diff
@@ -1,194 +0,0 @@
-#!/bin/bash
```
Comment (nabokihms) [inline shell_lib/hooks/storage_class_change.sh:1]: I accidently deleted these hooks.

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @nabokihms. I've started the workflow for label 'deploy/web/test'...

**Review by z9r5 — DISMISSED** [review]

**Review by konstantin-axenov — DISMISSED** [review]

File: modules/460-log-shipper/docs/USAGE.md [hunk 2]
```diff
@@ -199,3 +199,63 @@ spec:
...
+```
+
+Non-debug non-JSON logs:
+
+```yaml
+---
+apiVersion: deckhouse.io/v1alpha1
+kind: ClusterLoggingConfig
+metadata:
+  name: non-debeg-logs
```
Comment (konstantin-axenov) [inline modules/460-log-shipper/docs/USAGE.md:230]:
```suggestion
  name: non-debug-logs
```

File: modules/460-log-shipper/docs/USAGE_RU.md [hunk 3]
```diff
@@ -198,3 +198,63 @@ spec:
...
+```
+
+Не debug и не JSON-логи:
+
+```yaml
+---
+apiVersion: deckhouse.io/v1alpha1
+kind: ClusterLoggingConfig
+metadata:
+  name: non-debeg-logs
```
Comment (konstantin-axenov) [inline modules/460-log-shipper/docs/USAGE_RU.md:229]:
```suggestion
  name: non-debug-logs
```

**Review by yalosev — APPROVED** [review]

**Pull request merged**